[Tjan Eng Ger, Kevin] iP

Nephelite opened the pull request:
### SuiseiDuke ☄️ 
>So I say, what I wanted to be was never
>A Cinderella who waits forever
>But the prince who goes and gets things done
>Just like the shooting star I have become - Hoshimachi Suisei, [Stellar Stellar](https://www.youtube.com/watch?v=a51VH9BYzZA)

![image](https://user-images.githubusercontent.com/97427604/187021548-b9dc74aa-b07f-447f-99e5-8eb3a3727970.png)

SuiseiDuke ☄️ helps you get things done faster by helping you remember things you need to do. It's
- text-based
- simple to learn
- efficient with no fluff
- _ultra-fast_

All you need to do is to
1. Download it
2. Start it up
3. Add your tasks
4. Let it manage your tasks for you

And it's absolutely **_FREE_**!

Features:
- [x] Add Tasks (ToDos, Deadlines, Events)
- [x] Delete Tasks
- [x] Mark tasks as done/not done
- [x] List your tasks
- [x] Help command
- [ ] Tagging (Coming soon...)
---
SuiseiDuke ☄️ can also help you practice Java. Here is the `` main `` method:
```java
public static void main(String[] args) {
    Duke duke = new Duke();
    duke.run();
}
```

Nephelite committed:
Level 1: Greet, Echo, Exit

Nephelite committed:
Level 2: Add, List

Nephelite committed:
Level 3: Mark as Done

Nephelite committed:
Level4: ToDo, Event, Deadline

Nephelite committed:
A-TextUITesting

Nephelite committed:
Level 5: Handle Errors

Nephelite committed:
Level 6: Delete

Nephelite committed:
A-Enums, and custom command help

Nephelite committed:
Formatting and bug fixing

Nephelite committed:
More formatting (Most recent level is 6/A-Enum)

Nephelite committed:
Add save and load functionality to Duke

Nephelite committed:
Add a minimal version of Dates and Times

Nephelite committed:
Merge remote-tracking branch 'origin/branch-Level-7'

Nephelite committed:
Merge remote-tracking branch 'origin/branch-Level-8'

Nephelite committed:
Add OOP structure to Duke

Nephelite committed:
Sort all class structures into packages

Nephelite committed:
Add Gradle support and a minor code style fixes

Nephelite committed:
Add simple JUnit test to Parser and ToDo

Nephelite committed:
Actually add the new JUnit files

Nephelite committed:
Convert code to comply with CS2103T standards

Nephelite committed:
Modify Gradle build files to properly use shadowJar

Nephelite committed:
Add Javadocs

Nephelite committed:
Add Find command

Nephelite committed:
Merge branch 'branch-A-CodingStandard'

Nephelite committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/HelpCommand.java

Nephelite committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/Command.java

Nephelite committed:
Finalise merge conflicts and add Javadocs for Find

Nephelite committed:
Remove irrelevant files

Nephelite committed:
Restucture save of task changes

Nephelite committed:
Update all JavaDocs

Nephelite committed:
Add more JavaDoc improvements

Nephelite committed:
Add GUI to Duke

Nephelite committed:
Update read.md to prepare for GUI

Nephelite committed:
Merge branch 'branch-Level-10'

cheeheng commented on config/checkstyle/checkstyle.xml:
```diff
@@ -0,0 +1,434 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
It is great that you used checkstyle to ensure coding standards are met.

cheeheng commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,51 @@
+    public abstract String execute(Storage storage) throws DukeException;
+
+    /**
+     * Checks if the command leads to the end of the current Duke session
```
Could you add an empty line between description and parameter section? I saw this in several other areas of the code. Reference: https://se-education.org/guides/conventions/java/intermediate.html - "Empty line between description and parameter section"

Nephelite replied:
Oh whoops, I did not notice that was a part of the convention. Thanks!

cheeheng commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,72 @@
+     * {@inheritDoc}
+     * @return true if the command ends the current session. Otherwise, false.
+     * @since 0.1
+      */
```
Should this `*/` be well-aligned? Reference: https://se-education.org/guides/conventions/java/intermediate.html - "Subsequent `*` is aligned with the first one"

Nephelite replied:
Whoops, nice catch there. Thanks!

cheeheng commented on src/main/java/duke/command/HelpCommand.java:
```diff
@@ -0,0 +1,68 @@
+        System.out.println("These are the commands I know.");
+        for (RecognisedCommand e : RecognisedCommand.values()) {
+            switch (e) {
+            case BYE:
```
I liked how you followed the coding standard for the switch clause.

cheeheng commented on src/main/java/duke/core/Duke.java:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * Storage object used to load any previous tasks Duke was tracking in a previous session.
+     */
+    private final Storage storage;
```
I like how you included comments even for private variables.

cheeheng commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,40 @@
+
+public class ParserTest {
+    @Test
+    public void inputToDo_toDoHello_toDoCommandReturned() {
```
I like how the JUnit test method convention was followed in the code.

cheeheng reviewed (commented):
I like how you put in effort to ensure that coding standards are fulfilled. Could you fix the isolated case where JavaDoc `*` was not well-aligned and add a space between the description and the parameter in the JavaDocs as suggested?

Nephelite committed:
Edit JavaDocs with cheeheng's suggested changes

Nephelite commented:
Thanks for your review! I have already implemented the changes to JavaDocs that you have suggested.

Nephelite committed:
Add proper exit program function

Nephelite committed:
Assert: Add assert statements to guarantee assumptions

There are many assumptions in the code.

Assert statements help to detect if assumptions made do not hold

Adding assert statements allow easy checks for any potential logical
errors made.

Let's add asserts to:
*duke.task.Task
*duke.util.Storage
*duke.util.TaskList
*commands in the duke.command package (Deadline, Delete, Event,
Find, List, Mark, ToDo, Unmark)

Nephelite committed:
Code Quality: Improve readability of code

Code abstraction levels uneven, particularly in command classes

Code is difficult to understand without some time

Implement SLAP principles to make code more readable

Let's add SLAP principles to all command classes where possible.
Additionally, add a method in Parser to parse for taskId to assist
the SLAP principles.

Nephelite committed:
Merge pull request #1 from Nephelite/branch-A-Assertions

Assert: Add assert statements to guarantee assumptions

Nephelite committed:
Merge pull request #2 from Nephelite/branch-A-CodeQuality

Code Quality: Improve readability of code

Nephelite committed:
TagCommand: Add command to tag a task

Tasks in list can be hard to read

Tasks may have elements not included in description that users can
use to differentiate between them

Adding tags allows users to easily identify tasks within the task
list.

Let's add a TagCommand to allow users to tag tasks with unique
characteristics. Storage also saves and loads tags. Find also
searches both task description and tags.

Nephelite committed:
Merge pull request #3 from Nephelite/branch-C-Tagging

TagCommand: Add command to tag a task

Nephelite committed:
Add background image to scroll pane and color to text box

Nephelite committed:
Improve GUI to have alternate colors and new user icon

Nephelite committed:
Give credit for reused work

Nephelite committed:
Add title to Application window

Nephelite committed:
Update User Guide

Nephelite committed:
Amendment to javadocs

Nephelite committed:
Ready for release

Nephelite committed:
Add link to release to README.md

Nephelite committed:
Make Storage compatible with Unix

Nephelite committed:
Fix style issues and update README format issues

Nephelite committed:
Fix Readme format

Nephelite committed:
Fix typo in error message

Nephelite committed:
Change image files for Linux compatibility

Nephelite committed:
Disable resizing and credited changes to build.gradle

Nephelite committed:
Update plugin version

Nephelite committed:
Update README
